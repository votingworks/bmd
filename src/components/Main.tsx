import styled from 'styled-components'

interface Props {
  noMargin?: boolean
}

const Main = styled('main')<Props>`
  display: flex;
  flex-direction: column;
  flex: 1;
  margin: ${({ noMargin }: Props) => (noMargin ? undefined : '1rem')};
  @media print {
    justify-content: flex-start;
    margin: 0;
  }
`

interface ChildProps {
  center?: boolean
  centerVertical?: boolean
  centerHorizontal?: boolean
}

export const MainChild = styled('div')<ChildProps>`
  width: 100%;
  max-width: 40rem;
  margin: ${({
    center = false,
    centerVertical = center,
    centerHorizontal = true,
  }: ChildProps) =>
    `${centerVertical ? 'auto' : '0'} ${centerHorizontal ? 'auto' : '0'}`};
`

export default Main
