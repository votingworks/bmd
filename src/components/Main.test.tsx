import { axe } from 'jest-axe'
import React from 'react'
import { render } from 'react-testing-library'

import Main, { MainChild } from './Main'

it(`renders Main with child`, async () => {
  const { container } = render(
    <Main>
      <MainChild>foo</MainChild>
    </Main>
  )
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders centered content`, async () => {
  const { container } = render(
    <Main>
      <MainChild center>foo</MainChild>
    </Main>
  )
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders not centered content`, async () => {
  const { container } = render(
    <Main>
      <MainChild centerHorizontal={false}>foo</MainChild>
    </Main>
  )
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})
