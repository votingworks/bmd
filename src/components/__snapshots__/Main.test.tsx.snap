// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Main with child 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 1rem;
}

.c1 {
  width: 100%;
  max-width: 40rem;
  margin: 0 auto;
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 0;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    foo
  </div>
</main>
`;

exports[`renders centered content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 1rem;
}

.c1 {
  width: 100%;
  max-width: 40rem;
  margin: auto auto;
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 0;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    foo
  </div>
</main>
`;

exports[`renders not centered content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 1rem;
}

.c1 {
  width: 100%;
  max-width: 40rem;
  margin: 0 0;
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 0;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    foo
  </div>
</main>
`;
